client/my-sites/stats/hooks/use-intervals.tsx

```diff
@@ -56,9 +57,16 @@ function useIntervals( siteId: number | null ): IntervalsType {
 				},
 			};
 		}, {} );
-	} );
+	},
+	Object.values( intervals ).map(
+		( { statType } ) =>
+			( state: object, siteId ) =>
+				shouldGateStats( state, siteId, statType )
+	)
+);
```

Comment:
One idea that I believe would simplify all related code: instead of decorating the `intervals` array with the `isGated` field for each item, with all the memoization issues that come from it, let's calculate the `isGated` value only at the places that need it, using a hook. These places all want only an `isGated` value for a single `statType`, not for all of them.

Instead of this:
```js
function StatsOption( { interval } ) {
  if ( interval.isGated ) { /* show gated icon */ }
}
```
do this:
```js
function StatsOption( { siteId, interval } ) {
  const isGated = useGatedInterval( siteId, interval.statType );
  if ( interval.isGated ) { /* show gated icon */ }
}
```

Reply:
@Automattic/red is this something you folks wish to try out?

Reply:
Looks like an elegant solution. Thanks 👍